lib/components/dashboard/degree-info/DistributionBarsJSX.tsx

```diff
@@ -192,39 +192,44 @@ const DistributionBarsJSX: FC<{ major: Major }> = ({ major }) => {
       }
     }
   };
+
   const updateReqs = (reqs: [string, requirements[]][], courseObj) => {
-    let inNonExclusive: boolean = false;
-    // Exclusive check:
-    // If the requirement is exclusive, this means that if a course fulfills the requirement,
-    // it cannot fulfill any other requirements. Alternatively, if a course fulfills any other requirement, it cannot fulfill this one.
-    reqs.forEach((reqGroup, i) =>
-      reqGroup[1].forEach((req: requirements, j: number) => {
-        if (
-          courseObj !== null &&
-          checkRequirementSatisfied(req, courseObj) &&
-          (req.exclusive === undefined || !req.exclusive) &&
-          (req.fulfilled_credits < req.required_credits ||
-            (req.required_credits === 0 && req.fulfilled_credits === 0))
-        ) {
-          reqs[i][1][j].fulfilled_credits += parseInt(courseObj.credits);
-          inNonExclusive = true;
-        }
-      }),
-    );
-    reqs.forEach((reqGroup, i) =>
-      reqGroup[1].forEach((req: requirements, j: number) => {
-        if (
-          courseObj !== null &&
-          ((checkRequirementSatisfied(req, courseObj) &&
-            req.exclusive &&
-            req.fulfilled_credits < req.required_credits &&
-            !inNonExclusive) ||
-            (req.required_credits === 0 && req.fulfilled_credits === 0))
-        ) {
-          reqs[i][1][j].fulfilled_credits += parseInt(courseObj.credits);
-        }
-      }),
-    );
+    // exclusive check:
+    // If exclusive is undefined, the course may double count for any number of distributions
+    // If exclusive is string[], the specified distributions / fine requirements are 'whitelisted'
+    // When a course satisfies a distribution, exclusive takes the value of distribution.exclusive
+    if (!courseObj) return;
+    let distExclusive: string[] | undefined = undefined; // initial value
+    reqs.forEach((reqGroup, i) => {
```

Comment:
Trivial comment: this is O(N^2). Is there a better way to do this?

Reply:
I don't think so, because reqs is a 2d array :? Is there a something I'm missing? 

<img width="710" alt="Screen Shot 2022-07-23 at 9 02 23 PM" src="https://user-images.githubusercontent.com/90944895/180604013-95c2360f-ae4c-4998-88c9-86372eb3bb5f.png">